pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1622,6 +1631,27 @@ func (s *AWSCloud) listSubnetIDsinVPC(vpcId string) ([]string, error) {
 	return subnetIds, nil
 }
 
+func isSubnetPublic(rt []*ec2.RouteTable, subnetID string) bool {
+	for _, table := range rt {
+		var found bool
+		for _, assoc := range table.Associations {
+			if orEmpty(assoc.SubnetId) == subnetID {
+				found = true
+				break
+			}
+		}
+		if !found {
+			continue
+		}
+		for _, route := range table.Routes {
+			if strings.HasPrefix(orEmpty(route.GatewayId), "igw") {
```

Comment:
This is a scary way of determining whether a subnet is public, but I haven't found anything better.  A comment to say what we're doing would be great.  Something like "a public subnet is one which has an internet gateway route; there's no direct way in the AWS API to determine that fact; we look for the gatewayId and make sure it has the prefix of igw to differentiate from the default in-subnet route which is called "local".

But you understand this better than I do, so I hope your comment will be better :-)